local/pipeline_test.go

```diff
@@ -209,12 +209,12 @@ func TestPipelineParsing(t *testing.T) {
 							"echo hello world",
 							"pwd"
 						],
-						"plugins": {
-							"blah-blah/blah#v0.0.1": null,
-							"blah-blah/another#v0.0.1": {
+						"plugins": [
+							{"blah-blah/blah#v0.0.1": null},
+							{"blah-blah/another#v0.0.1": {
 								"my_config":"totes"
-							}
-						}
```

Comment:
From what I recall of what you found out when switching the YAML parser on the agent, the unordered map YAML syntax is secretly ordered when the agent parses. So it is the case that JSON syntax being removed here is not something the agent would send to the API?

Reply:
I didn't really want to spend the afternoon fixing the pipeline parser here - it would be a lot more involved.

The previous JSON is exactly what the agent would send if the plugins were all in one map. The new JSON is what it would send if the plugins were (more correctly) in a sequence of one-item maps.

Maybe the agent should convert the first form into the second?

Reply:
Ok makes sense. So the JSON parsing is not quite the same in the CLI as the API in that it won't preserve order in an unordered object 🙉 It's probably worth recording that as a ticket in the backlog, or even an issue in this repo with some context.

> Maybe the agent should convert the first form into the second?

I think that's prudent, but might not be necessary. I suspect it "just works" in the API because something something insertion order preserving iteration for hashes in ruby.

It would be good to make this explicit in all 3 of the API, CLI and agent though.

Reply:
> It would be good to make this explicit in all 3 of the API, CLI and agent though.

Yes. And if it were up to me, I'd make it so we never accept a JSON object or YAML map where we care about the order.

Reply:
We can only dream...for now